File: app/src/main/java/com/neonide/studio/app/NeonIDEApplication.java
package com.neonide.studio.app;

import android.content.Context;
import com.termux.app.TermuxApplication;
import com.neonide.studio.logger.IDEFileLogger;
import com.termux.shared.logger.Logger;

public class NeonIDEApplication extends TermuxApplication {

    @Override
    public void onCreate() {
        super.onCreate();
        
        Context context = getApplicationContext();
        
        // Init sora-editor file providers
        io.github.rosemoe.sora.langs.textmate.registry.FileProviderRegistry.getInstance().addFileProvider(new io.github.rosemoe.sora.langs.textmate.registry.provider.AssetsFileResolver(context.getAssets()));

        // Register IDEFileLogger to capture all logs sent via Logger class
        Logger.setExternalLogger((priority, tag, message) -> {
            IDEFileLogger.log(context, tag + ": " + message);
        });
    }
}

File: app/src/main/java/com/neonide/studio/app/NeonIDEApplication.kt
package com.neonide.studio.app

import com.neonide.studio.logger.IDEFileLogger
import com.termux.app.TermuxApplication
import com.termux.shared.logger.Logger
import io.github.rosemoe.sora.langs.textmate.registry.FileProviderRegistry
import io.github.rosemoe.sora.langs.textmate.registry.provider.AssetsFileResolver

class NeonIDEApplication : TermuxApplication() {
    override fun onCreate() {
        super.onCreate()

        val context = applicationContext

        FileProviderRegistry.getInstance()
            .addFileProvider(AssetsFileResolver(context.assets))

        Logger.setExternalLogger { priority, tag, message ->
            IDEFileLogger.log(context, "$tag: $message")
        }
    }
}
